### ISSUE TITLE
Recover changes from old master branch

### ISSUE THREAD

**Issue by tillkruss** [predis/predis#624]
Is there anything we should recover from the stale master branch?

From #587: 

> The master branch is meant to become v2.0, but since it diverged quite a lot from v1.1 most fixes cannot be easily merged. At first I intended to maintain both v1.1 and master-to-be-v2.0 but in retrospect I guess that plan actually backfired so now the plan is to just fix v1.1 for both implementation bugs and compatiblity issues with newer PHP versions, maybe a few new Redis commands, and leave it at that. On the master branch, aside from applying the same fixes, we should drop support once and for all for PEAR and older PHP versions and rework parts of code that could benefit from new features introduced in PHP 7 before implementing anything new.

**Comment by AhmedRaafat14** [comment]
That is really a good question.
Personally, I have to check the differences and see, but mostly I think we just need to follow the roadmap and don't port allot of changes.

But yes we will see after we check :).

**Comment by nrk** [comment]
During the next few days I'll review every change I applied back then, but as far as I remember everything I pushed there had a reason to exist in the next major version. Just to make an example, one of the biggest changes in v2.0 is how commands are handled by default with a convention-over-configuration approach (see 62b421f, 4b47639 and 822f02b) which basically supersedes the whole concept of profiles. Maybe some things could be improved now that we'll finally drop older PHP versions, but the overall direction was indeed the right one.
